pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -856,6 +878,56 @@ func (r *ReconcileClusterDeployment) setupClusterInstallServiceAccount(namespace
 	return currentSA, nil
 }
 
+func selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
+		if cdName, ok := pod.Labels[install.ClusterDeploymentNameLabel]; ok {
+			retval = append(retval, reconcile.Request{NamespacedName: types.NamespacedName{
+				Name:      cdName,
+				Namespace: pod.Namespace,
+			}})
+		}
+	}
+	return retval
+}
+
+// Calculates the restart count for both hive and installer container for a cd and sets it to metricClusterDeploymentInstallRetriesTotal
+func (r *ReconcileClusterDeployment) updateInstallRetriesTotalMetric(clusterDeploymentName string, clusterDeploymentNamespace string, cdLog log.FieldLogger) {
+	labels := map[string]string{"cluster_deployment": clusterDeploymentName, "namespace": clusterDeploymentNamespace}
+	installerPodLabels := map[string]string{install.ClusterDeploymentNameLabel: clusterDeploymentName, install.InstallJobLabel: "true"}
+	currentRetriesMetric, err := metricClusterDeploymentInstallRetriesTotal.GetMetricWith(labels)
+	if err != nil {
+		cdLog.Error("Error getting cluster deployment install retries total metrics.")
+	} else {
+		cdLog.Infof("currentRetriesMetric:%v", currentRetriesMetric)
+		pods := &corev1.PodList{}
+		err = r.Client.List(context.Background(), client.MatchingLabels(installerPodLabels), pods)
+		if err != nil {
+			log.WithError(err).Error("error listing pods")
+		} else {
+			podRestarts := 0
+			for _, pod := range pods.Items {
+				cdLog.Infof("pod name:%v", pod.Name)
+				for _, cs := range pod.Status.ContainerStatuses {
+					cdLog.Infof("Container Name:%v", cs.Name)
+					cdLog.Infof("Pod Restarts:%v", cs.RestartCount)
```

Comment:
These should  not be logged at Info, but once you're happy it's working they can probably be removed entirely.